H2MLauncher.UI/MapColorConverter.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.Globalization;
+using System.IO;
+using System.Windows.Data;
+using System.Windows.Media;
+
+namespace H2MLauncher.UI
+{
+    public class MapColorConverter : IValueConverter
+    {
+        public object Convert(object value, Type targetType, object parameter, CultureInfo culture)
+        {
+            if (value is string folderName)
+            {
+                // Define the directory where map folders are located
+                string executableDirectory = AppDomain.CurrentDomain.BaseDirectory;
+                string mapsDirectory = Path.Combine(executableDirectory, "h2m-usermaps");
+
+                // Construct the full path to the folder
+                string folderPath = Path.Combine(mapsDirectory, folderName);
+
+                System.Diagnostics.Debug.WriteLine($"Checking folder: {folderPath}");
+
+                // Check if the folder exists
+                bool folderExists = Directory.Exists(folderPath);

```

Comment:
Right now, you are scanning the directory every time when you check if the map exists. It would be better to scan the directory once when the launcher opens and store the maps in a collection like a HashSet, then check against it which would be more efficient.

Reply:
I think you should entirely avoid doing anything like that in a converter, but instead move this logic to the view model by introducing a `UserHasMap` property and converting from true or false